src/go/rpk/pkg/config/params.go

```diff
@@ -284,6 +279,13 @@ func (p *Params) Load(fs afero.Fs) (*Config, error) {
 		if !errors.Is(err, afero.ErrFileNotFound) {
 			return nil, err
 		}
+		// If there is no file, we set the file location to the passed
+		// --config value, otherwise we use the default.
+		if p.ConfigPath != "" {
+			c.fileLocation = p.ConfigPath
+		} else {
+			c.fileLocation = DefaultConfigPath
+		}
```

Comment:
Shouldn't this be done outside of the readConfig block? I think if it isn't, then some messages might look weird -- e.g., in mode,

```go
fmt.Printf("Writing %q mode defaults to %q\n", mode, cfg.FileLocation())
```
wouldn't this print `""` for the location?

Reply:
Inside `readConfig` we set `cfg.fileLocation` if we successfully read a file, in this block we set `cfg.fileLocation` when the file is not yet on disk (either running on MacOS or setting an arbitrary path via --config flag):

```yaml
# File exists in default location
$ ./linux-amd64/rpk redpanda mode prod
Writing "prod" mode defaults to "/etc/redpanda/redpanda.yaml"

# File exist in arbitrary location (--config)
$ ./linux-amd64/rpk redpanda mode prod --config ~/Downloads/redpanda.yaml 
Writing "prod" mode defaults to "/home/rvasquez/Downloads/redpanda.yaml"

# File doesn't exist in arbitrary location (--config)
$ ./linux-amd64/rpk redpanda mode prod --config ~/Downloads/notexistent.yaml
Writing "prod" mode defaults to "/home/rvasquez/Downloads/notexistent.yaml"

$ cat ~/Downloads/notexistent.yaml 
redpanda:
    data_directory: /var/lib/redpanda/data
    node_id: 0
    seed_servers: []
    rpc_server:
        address: 0.0.0.0
        port: 33145
    kafka_api:
        - address: 0.0.0.0
          port: 9092
    admin:
        - address: 0.0.0.0
          port: 9644
rpk:
    tune_network: true
    tune_disk_scheduler: true
    tune_disk_nomerges: true
    tune_disk_write_cache: true
    tune_disk_irq: true
    tune_cpu: true
    tune_aio_events: true
    tune_clocksource: true
    tune_swappiness: true
    coredump_dir: /var/lib/redpanda/coredump
    tune_ballast_file: true
pandaproxy: {}
schema_registry: {}
```

Reply:
Oh I see, that makes sense. I get tripped up sometimes on the difference between Params and Config.